src/vsr/replica.zig

```diff
@@ -3547,10 +3559,31 @@ pub fn ReplicaType(
             assert(self.commit_stage == .checkpoint_grid_repair);
             assert(self.commit_prepare.?.header.op == self.op);
 
+            self.commit_dispatch(.checkpoint_free_set);
+        }
+
+        fn commit_op_checkpoint_free_set(self: *Self) void {
+            assert(self.commit_stage == .checkpoint_free_set);
+            assert(self.commit_prepare.?.header.op == self.op);
+            assert(self.superblock.free_set.opened);
+
+            self.superblock.free_set_encoded.checkpoint(commit_op_checkpoint_free_set_callback);
+        }
+
+        fn commit_op_checkpoint_free_set_callback(set: *SuperBlock.FreeSetEncoded) void {
+            const superblock = @fieldParentPtr(SuperBlock, "free_set_encoded", set);
+            const self = @fieldParentPtr(Self, "superblock", superblock);
+            assert(self.commit_stage == .checkpoint_free_set);
+            assert(self.commit_prepare.?.header.op == self.op);
+            assert(self.superblock.free_set.opened);
+            assert(self.superblock.free_set.count_released() ==
```

Comment:
In `commit_op_checkpoint_superblock_callback`, assert that `free_set.count_released() == 0`.

Reply:
Hm, I don't get this? In `commit_op_checkpoint_superblock_callback`, the free set had been checkpointed already, so `free_set.count_released()  == free_set_encoded.block_count`?

Reply:
Ah right, nevermind!